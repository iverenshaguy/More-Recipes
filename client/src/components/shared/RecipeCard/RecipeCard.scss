@import '../../../assets/scss/modules';

.recipe-card {
  .card {
    box-shadow: $recipe-card-shadow;
  }

  a, p, small:not(.text-muted) {
    color: initial;
    text-decoration: none;
  }

  &:hover {
    transform: scale(1.025) 0.5s;
  }

  .recipe-details {
    border-bottom: 1px solid $recipe-card-grey;
    border-top: 1px solid $recipe-card-grey;
    padding: 0.5rem 1rem;
  }

  .card-image {
    border: 0;
  }

  .card-image-overlay {
    padding-top: 17%;
    background-color: $transparent-middle;
  }

  img {
    width: 100% !important;
    height: 150px !important;
  }
}

.category-card {
  .card {
    box-shadow: $recipe-card-shadow;
  }

  a, p, small:not(.text-muted) {
    color: initial;
    text-decoration: none;
  }

  &:hover {
    transform: scale(1.025) 0.5s;
  }
}

.user-profile-cards {
  .card {
    box-shadow: $recipe-card-shadow;
  }
}

.dropdown-menu {
  box-shadow: $recipe-card-shadow;
}
