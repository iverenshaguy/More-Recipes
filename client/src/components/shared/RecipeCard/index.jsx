import React from 'react';
import { Link } from 'react-router-dom';
import { Card, CardImg, CardText, CardHeader, CardImgOverlay } from 'reactstrap';
import FontAwesome from 'react-fontawesome';
import { singleRecipePropTypes } from '../../../helpers/proptypes';
import './RecipeCard.scss';

const RecipeCard = ({ recipe }) => (
  <div className="col-xs-12 col-sm-6 col-md-4 col-lg-3 mt-4 recipe-card">
    <Card>
      <Link to={`/recipes/${recipe.id}`}>
        <CardHeader className="py-2">
          <CardText className="d-inline"><small>@{recipe.User.username}</small></CardText>
          <CardText className="d-inline float-right">
            <small className="text-muted"><FontAwesome name="heart" /> {recipe.upvotes}</small>
          </CardText>
        </CardHeader>
        <Card className="m-0 p-0 card-image">
          <CardImg
            bottom
            width="100%"
            className="img-fluid"
            src="images/jollof-rice-img.jpg"
            alt="recipe"
          />
          <CardImgOverlay className="text-center card-image-overlay text-white px-2 recipe-rating">
            <h5 className="mb-1 pt-3">{recipe.recipeName}</h5>
          </CardImgOverlay>
        </Card>
      </Link>
    </Card>
  </div>
);

RecipeCard.propTypes = {
  ...singleRecipePropTypes
};

export default RecipeCard;
