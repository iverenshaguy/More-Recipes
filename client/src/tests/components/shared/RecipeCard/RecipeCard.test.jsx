import React from 'react';
import RecipeCard from '../../../../components/shared/RecipeCard';

const recipe = {
  id: 8,
  recipeName: 'Chicken Pie',
  recipeImage: 'sjkks',
  totalTime: '1hr 30 minutes',
  difficulty: 'Easy',
  upvotes: 150,
  views: 290,
  rating: '4.760',
  User: {
    userId: 2,
    username: 'westsouth',
    profilePic: null
  }
};

describe('RecipeCard', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<RecipeCard recipe={recipe} />);
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.unmount();
  });
});
