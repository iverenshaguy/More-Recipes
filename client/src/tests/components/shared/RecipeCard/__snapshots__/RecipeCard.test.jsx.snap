// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeCard renders correctly 1`] = `
<div
  className="col-xs-12 col-sm-6 col-md-4 col-lg-3 mt-4 recipe-card"
>
  <Card
    tag="div"
  >
    <Link
      replace={false}
      to="/recipes/8"
    >
      <CardHeader
        className="py-2"
        tag="div"
      >
        <CardText
          className="d-inline"
          tag="p"
        >
          <small>
            @
            westsouth
          </small>
        </CardText>
        <CardText
          className="d-inline float-right"
          tag="p"
        >
          <small
            className="text-muted"
          >
            <FontAwesome
              name="heart"
            />
             
            150
          </small>
        </CardText>
      </CardHeader>
      <Card
        className="m-0 p-0 card-image"
        tag="div"
      >
        <CardImg
          alt="recipe"
          bottom={true}
          className="img-fluid"
          src="images/jollof-rice-img.jpg"
          tag="img"
          width="100%"
        />
        <CardImgOverlay
          className="text-center card-image-overlay text-white px-2 recipe-rating"
          tag="div"
        >
          <h5
            className="mb-1 pt-3"
          >
            Chicken Pie
          </h5>
        </CardImgOverlay>
      </Card>
    </Link>
  </Card>
</div>
`;
